File: cpp/n476_FindComplement.cpp
// https://leetcode-cn.com/problems/number-complement/
#include <iostream>
#include <vector>
using namespace std;

class Solution{
public:
    int FindComplement(int num){
        int res = 0;
        long base = 1;
        while(num > 0){
            int tmp = 1 - (num % 2);
            res += tmp * base;
            base *= 2;
            num /= 2;
        }
        return res;
    }
};

int main(){
    return 0;
}

File: python/n476_FindComplement.py
# https://leetcode-cn.com/problems/number-complement/
class Solution:
    def findComplement(self, num: int) -> int:
        res = 0
        base = 1
        while num > 0:
            tmp = 1 - (num%2)
            res += (tmp * base)
            base *= 2
            num //= 2
        return res


if __name__ == "__main__":
    pass      